TestWTF tests compilation unit names clash with WTF implementation compilation unit names, causing linker warnings

emw-apple opened the pull request:
#### ce7e7da9c4dc1f34865b5e242364bea28dbf9eb5
<pre>
TestWTF tests compilation unit names clash with WTF implementation compilation unit names, causing linker warnings
<a href="https://bugs.webkit.org/show_bug.cgi?id=224860">https://bugs.webkit.org/show_bug.cgi?id=224860</a>
rdar://problem/77255847

Reviewed by Alexey Proskuryakov.

libWTF.a was being merged into libTestWTF.a, causing a
naming collision that resulted in thousands of warnings like

    Libtool libTestWTF
    /Volumes/.../usr/bin/libtool: warning same member name (AtomString.o) in output file used for input files:

and

    GenerateDSYMFile TestWTF.dSYM
    warning: (x86_64)  could not find object file symbol for symbol __ZNK3WTF10AtomString23convertToASCIILowercaseEv

Instead, move the linkage of libWTF.a down into the TestWTF binary.
Tests in TestWTF can have the same file name as their implementation in
WTF without symbol name collisions, because they are never part of the
same static archive.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/267099@main">https://commits.webkit.org/267099@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d9a883d1c426c2198c292977f19599f6ec5c716

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15531 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15837 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16200 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17287 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14566 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15703 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18354 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15932 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17131 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15716 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16152 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13207 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18031 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13420 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14009 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20941 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14472 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14175 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17447 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14762 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12506 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14018 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3747 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18380 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14582 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Can't meaningfully review the pbxproj change, but the explanation makes good sense, and the XML diff for it would indeed look something like this.

emw-apple committed:
TestWTF tests compilation unit names clash with WTF implementation compilation unit names, causing linker warnings
https://bugs.webkit.org/show_bug.cgi?id=224860
rdar://problem/77255847

Reviewed by Alexey Proskuryakov.

libWTF.a was being merged into libTestWTF.a, causing a
naming collision that resulted in thousands of warnings like

    Libtool libTestWTF
    /Volumes/.../usr/bin/libtool: warning same member name (AtomString.o) in output file used for input files:

and

    GenerateDSYMFile TestWTF.dSYM
    warning: (x86_64)  could not find object file symbol for symbol __ZNK3WTF10AtomString23convertToASCIILowercaseEv

Instead, move the linkage of libWTF.a down into the TestWTF binary.
Tests in TestWTF can have the same file name as their implementation in
WTF without symbol name collisions, because they are never part of the
same static archive.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/267099@main
